### PR TITLE
Upgradability mechanisms script version

### PR DESCRIPTION
**Pull request by talelbaz**
Changes type of block version, txn version, and scriptPublicKey version to uint16.
Also, changes the tests to suit.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,7 +6,7 @@ require (
 	github.com/btcsuite/go-socks v0.0.0-20170105172521-4720035b7bfd
 	github.com/btcsuite/winsvc v1.0.0
 	github.com/davecgh/go-spew v1.1.1
-	github.com/golang/protobuf v1.4.2
+	github.com/golang/protobuf v1.4.3
```
Comment (stasatdaglabs) [inline go.mod:9]: Please revert this.

File: infrastructure/network/netadapter/server/grpcserver/protowire/messages.proto [hunk 2]
```diff
@@ -693,9 +697,14 @@ message RpcTransactionInput {
   uint64 sequence = 3;
 }
 
+message RpcScriptPubKey {
```
Comment (stasatdaglabs) [inline infrastructure/network/netadapter/server/grpcserver/protowire/messages.proto:700]: Please rename this to RpcScriptPublicKey.

File: infrastructure/network/netadapter/server/grpcserver/protowire/messages.proto [hunk 3]
```diff
@@ -693,9 +697,14 @@ message RpcTransactionInput {
...
+  uint32 version = 1;
+  string scriptPubKey = 2;
```
Comment (stasatdaglabs) [inline infrastructure/network/netadapter/server/grpcserver/protowire/messages.proto:702]: Please rename this to scriptPublicKey.

File: infrastructure/network/netadapter/server/grpcserver/protowire/messages.proto [hunk 4]
```diff
@@ -693,9 +697,14 @@ message RpcTransactionInput {
...
+}
+
 message RpcTransactionOutput {
   uint64 amount = 1;
-  string scriptPubKey = 2;
+  RpcScriptPubKey scriptPubKey = 2;
```
Comment (stasatdaglabs) [inline infrastructure/network/netadapter/server/grpcserver/protowire/messages.proto:707]: Please rename this to scriptPublicKey.

File: infrastructure/network/netadapter/server/grpcserver/protowire/messages.proto [hunk 5]
```diff
@@ -705,7 +714,7 @@ message RpcOutpoint {
 
 message RpcUtxoEntry {
   uint64 amount = 1;
-  string scriptPubKey = 2;
+  RpcScriptPubKey scriptPubKey = 2;
```
Comment (stasatdaglabs) [inline infrastructure/network/netadapter/server/grpcserver/protowire/messages.proto:717]: Please rename this to RpcScriptPublicKey.

File: testing/integration/utxo_index_test.go [hunk 6]
```diff
@@ -132,7 +133,22 @@ func TestUTXOIndex(t *testing.T) {
 			t.Fatalf("Missing entry in UTXOs response: %s:%d",
 				notificationEntry.Outpoint.TransactionID, notificationEntry.Outpoint.Index)
 		}
-		if *notificationEntry.UTXOEntry != *foundResponseEntry.UTXOEntry {
+		if notificationEntry.UTXOEntry.Amount != foundResponseEntry.UTXOEntry.Amount {
```
Comment (stasatdaglabs) [inline testing/integration/utxo_index_test.go:136]: Please utxo UTXOEntry.Equal instead.

**Review by stasatdaglabs — CHANGES_REQUESTED** [review]

**Comment by stasatdaglabs** [comment]
Rebuild after update in master

File: app/rpc/rpccontext/verbosedata.go [hunk 7]
```diff
@@ -134,7 +134,7 @@ func (ctx *Context) buildTransactionVerboseInputs(tx *externalapi.DomainTransact
 		// The disassembled string will contain [error] inline
 		// if the script doesn't fully parse, so ignore the
 		// error here.
-		disbuf, _ := txscript.DisasmString(constants.MaximumScriptPublicKeyVersion, transactionInput.SignatureScript)
+		disbuf, _ := txscript.DisasmString(constants.ScriptPublicKeyVersion, transactionInput.SignatureScript)
```
Comment (someone235) [inline app/rpc/rpccontext/verbosedata.go:137]: Change to MaxScriptPublicKeyVersion

File: domain/consensus/utils/constants/constants.go [hunk 8]
```diff
@@ -5,10 +5,13 @@ import "math"
 const (
 	// BlockVersion represents the current version of blocks mined and the maximum block version
 	// this node is able to validate
-	BlockVersion = 0
+	BlockVersion uint16 = 0
 
 	// TransactionVersion is the current latest supported transaction version.
-	TransactionVersion = 0
+	TransactionVersion uint16 = 0
```
Comment (someone235) [inline domain/consensus/utils/constants/constants.go:11]: Add Max

File: domain/consensus/utils/constants/constants.go [hunk 9]
```diff
@@ -5,10 +5,13 @@ import "math"
 const (
 	// BlockVersion represents the current version of blocks mined and the maximum block version
 	// this node is able to validate
-	BlockVersion = 0
+	BlockVersion uint16 = 0
```
Comment (someone235) [inline domain/consensus/utils/constants/constants.go:8]: Add Max

File: domain/consensus/utils/txscript/sign.go [hunk 10]
```diff
@@ -188,29 +188,30 @@ func (sc ScriptClosure) GetScript(address util.Address) ([]byte, error) {
...
 		return nil, err
 	}
-	// TODO: Do not merge it!!
-	scriptpubKey := &externalapi.ScriptPublicKey{
-		Script:  sigScript,
-		Version: scriptPubKey.Version,
-	}
+
 	if class == ScriptHashTy {
+		signedScriptPublicKey := &externalapi.ScriptPublicKey{
```
Comment (someone235) [inline domain/consensus/utils/txscript/sign.go:208]: Rename to scriptHashPreimageScriptPublicKey

File: domain/consensus/utils/txscript/standard_test.go [hunk 11]
```diff
@@ -323,48 +323,62 @@ func TestPayToAddrScript(t *testing.T) {
...
+
+		var scriptPublicKeyScript []byte
+		var scriptPublicKeyVersion uint16
+		if scriptPublicKey != nil {
+			scriptPublicKeyScript = scriptPublicKey.Script
+			scriptPublicKeyVersion = scriptPublicKey.Version
 		}
-		if !bytes.Equal(scriptPubKey.Script, expected) {
+
+		expectedScript := mustParseShortForm(test.expectedScript)
```
Comment (someone235) [inline domain/consensus/utils/txscript/standard_test.go:372]: `mustParseShortForm` should also get version

**Review by someone235 — CHANGES_REQUESTED** [review]

**Review by stasatdaglabs — APPROVED** [review]

**Review by someone235 — APPROVED** [review]

**Comment by stasatdaglabs** [comment]
Rebuild after update in master

**Pull request merged**